sycl/test/extensions/properties/mock_compile_time_properties.hpp

```diff
@@ -0,0 +1,140 @@
+
+#pragma once
+
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+struct bar_key {
+  using value_t = property_value<bar_key>;
+};
+
+struct baz_key {
+  template <int K>
+  using value_t = property_value<baz_key, std::integral_constant<int, K>>;
+};
+
+struct boo_key {
+  template <typename... Ts> using value_t = property_value<boo_key, Ts...>;
+};
+
+struct foo {
+  foo(int v) : value(v) {}
+  int value;
+};
+
+inline bool operator==(const foo &lhs, const foo &rhs) {
+  return lhs.value == rhs.value;
+}
+inline bool operator!=(const foo &lhs, const foo &rhs) { return !(lhs == rhs); }
+
+struct foz {
+  foz(float v1, bool v2) : value1(v1), value2(v2) {}
+  // Define copy constructor to make foz non-trivially copyable
+  foz(const foz &f) {
+    value1 = f.value1;
+    value2 = f.value2;
+  }
+  float value1;
+  bool value2;
+};
+
+inline bool operator==(const foz &lhs, const foz &rhs) {
+  return lhs.value1 == rhs.value1 && lhs.value2 == rhs.value2;
+}
+inline bool operator!=(const foz &lhs, const foz &rhs) { return !(lhs == rhs); }
+
+struct fir {
+  fir(float v1, bool v2) : value1(v1), value2(v2) {}
+  // Define copy constructor to make foz non-trivially copyable
+  fir(const foz &f) {
+    value1 = f.value1;
+    value2 = f.value2;
+  }
+  float value1;
+  bool value2;
+};
+
+inline bool operator==(const fir &lhs, const fir &rhs) {
+  return lhs.value1 == rhs.value1 && lhs.value2 == rhs.value2;
+}
+inline bool operator!=(const fir &lhs, const fir &rhs) { return !(lhs == rhs); }
+
+inline constexpr bar_key::value_t bar;
+template <int K> inline constexpr baz_key::value_t<K> baz;
+template <typename... Ts> inline constexpr boo_key::value_t<Ts...> boo;
+
+using foo_key = foo;
+using foz_key = foz;
+using fir_key = fir;
+
+template <> struct is_property_key<bar_key> : std::true_type {};
+template <> struct is_property_key<baz_key> : std::true_type {};
+template <> struct is_property_key<boo_key> : std::true_type {};
+template <> struct is_property_key<foo_key> : std::true_type {};
+template <> struct is_property_key<foz_key> : std::true_type {};
+template <> struct is_property_key<fir_key> : std::true_type {};
+
+template <typename syclObjectT>
+struct is_property_key_of<bar_key, syclObjectT> : std::true_type {};
+template <typename syclObjectT>
+struct is_property_key_of<baz_key, syclObjectT> : std::true_type {};
+template <typename syclObjectT>
+struct is_property_key_of<boo_key, syclObjectT> : std::true_type {};
+template <typename syclObjectT>
+struct is_property_key_of<foo_key, syclObjectT> : std::true_type {};
+template <typename syclObjectT>
+struct is_property_key_of<foz_key, syclObjectT> : std::true_type {};
+template <typename syclObjectT>
+struct is_property_key_of<fir_key, syclObjectT> : std::true_type {};
+
+namespace detail {
+template <> struct PropertyToKind<bar_key> {
+  static constexpr PropKind Kind =
+      static_cast<enum PropKind>(PropKind::PropKindSize + 0);
```

Comment:
I'm guessing the awkward `PropKindSize + X` constructs here is because we aren't actually making modifications to `property.hpp`, correct?

Reply:
Exactly. These properties are not "real" so we make their `PropKind` outside the usual range to avoid confusing them with actual properties.